Patches chap 2

dhurlburtusa opened the pull request:
I just finished with chap 2. Very educational!

dhurlburtusa committed:
Remove extra word

dhurlburtusa committed:
Fix pluralization

Commit:
Add missing word

Commit:
Mark as new word

Commit:
Remove awkward wording

Commit:
Remove commas
First comma not needed because it is not separating independent clauses.
Second comma not needed.

Commit:
Not mark as new word
Already seen in chap 1.

Commit:
Fix word

Commit:
Fix coordinate

Commit:
Fix typos

Commit:
Fix typos

Commit:
Merge branch 'patches' into patches-chap-2

Commit:
Fix pluralization

Commit:
Remove comma
Seems unnecessary

Commit:
Fix typo

Commit:
Fix typo

Commit:
Remove extraneous word

Commit:
Fix typos

Commit:
Fix reference

Commit:
Add missing word

Commit:
Revert some suggestions

Commit:
Merge branch 'patches' into patches-chap-2

Commit:
Remove parameter not wanted in the pseudocode

Commit:
Add missing word

Commit:
Fix typos

Commit:
Revert a suggestion

Commit:
Remove extra comma

Commit:
Remove repeated words

Commit:
Remove extra word

Commit:
Fix method name

Commit:
Remove extra comma

dhurlburtusa committed:
Merge branch 'davidjeck:main' into patches-chap-2

Commit:
Fix whitespace

Commit:
Remove extraneous word

Commit:
Remove extraneous word

davidjeck commented:
These all look great, except maybe for the commas that you want to remove.  I use commas (or parentheses or dashes) when I think they improve readability, even if they are not quite grammatically correct.  You might look at them and see whether you agree with me.

But I think you are right about the removed comma in intro.html.

Commit:
Remove extraneous word

Commit:
Revert some suggestions

Commit:
Put comma back in intended place

Commit:
Revert some suggestions

dhurlburtusa commented:
I reverted the comma changes except the ones in intro.xml.

I also fixed "the the". Thanks for finding it. What was the reg exp that you used? I have one that works across multiple lines: `(\b\w+)[\s\n]+[\\\*]*\s*\1\b` for some specific cases. Seems to work well in VS Code.
